public/pages/Dashboard/Container/DashboardOverview.tsx

```diff
@@ -204,10 +227,8 @@ export function DashboardOverview(props: OverviewProps) {
     history.replace({
       ...location,
       search: queryString.stringify(updatedParams),
-    })
-    if (dataSourceEnabled ? MDSOverviewState.selectedDataSourceId : true) {
```

Comment:
Does removing this conditional change the logic of `initializeDetectors()`?

Reply:
No, I didn't change any logic inside `initializeDetectors()`, why would it change the logic of it? I explained the reason why removing this conditional check here - https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/722#discussion_r1569214460